src/Mono.Android/Java.Interop/JavaConvert.cs

```diff
@@ -372,42 +372,39 @@ public static T FromJavaObject<T>(IJavaObject? value, out bool set)
 			return LocalJniHandleConverters [typeof (Android.Runtime.JavaObject)];
 		}
 
+		internal static IntPtr ToLocalJniHandle (object? value)
+		{
+			if (value == null) {
+				return IntPtr.Zero;
+			}
+			if (value is IJavaObject v) {
+				return JNIEnv.ToLocalJniHandle (v);
+			}
+			Func<object, IntPtr> converter = GetLocalJniHandleConverter (value, value.GetType ());
+			return converter (value);
+		}
+
 		public static TReturn WithLocalJniHandle<TValue, TReturn>(TValue value, Func<IntPtr, TReturn> action)
 		{
-			IntPtr lref = IntPtr.Zero;
+			IntPtr lref = ToLocalJniHandle (value);
 			try {
-				var v = value as IJavaObject;
-				if (v != null) {
-					lref = JNIEnv.ToLocalJniHandle (v);
-					return action (lref);
-				}
-				Func<object, IntPtr> converter = GetLocalJniHandleConverter (value, typeof (TValue));
```

Comment:
With this change, I think we can make `GetLocalJniHandleConverter<T>()` *non-generic*.  We don't have any benefit to a generic `GetLocalJniHandleConverter()`.

Since `converter` is a `Func<object, IntPtr>`, boxing can't be avoided, so…